### PR TITLE
[RLlib] DreamerV3: Make 200M (XL model) work; mixed float16 option

### PR DESCRIPTION
**Pull request by sven1977**
DreamerV3: Make 200M (XL model) work; mixed float16 option

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/dreamerv3/dreamerv3.py [hunk 1]
```diff
@@ -612,7 +643,7 @@ def training_step(self) -> ResultDict:
                 # update.
                 with self._timers["critic_ema_update"]:
                     self.learner_group.additional_update(
-                        timestep=self._counters[NUM_ENV_STEPS_TRAINED],
+                        timestep=self._counters[NUM_ENV_STEPS_SAMPLED],
```
Comment (sven1977) [inline rllib/algorithms/dreamerv3/dreamerv3.py:646]: Really important change. I think most algos should count by env steps. This is the most common metric used in the literature when reporting results over time for RL.

**Review by kouroshHakha — APPROVED** [review]

**Review by krfricke — APPROVED** [review]
stamp

**Review by zhe-thoughts — APPROVED** [review]
Yes this is critical PR to merge

**Comment by zhe-thoughts** [comment]
@sven1977 @krfricke @kouroshHakha : could one of you confirm that CI is OK, before I merge?

**Review by maxpumperla — APPROVED** [review]

**Comment by sven1977** [comment]
Hey @zhe-thoughts , feel free to merge this one here any time. Its tests have all passed and it's good to go.
Thanks! :)

**Comment by sven1977** [comment]
Hey @zhe-thoughts , this would be good to merge. All tests are passing now. Thank you so much!

**Pull request merged**